tethys_apps/cli/scaffold_templates/app_templates/default/tethysapp/+project+/app.py_tmpl

```diff
@@ -16,6 +18,11 @@ class {{class_name}}(TethysAppBase):
     tags = '{{tags}}'
     enable_feedback = False
     feedback_emails = []
+    analytics = bool('analytical' in settings.INSTALLED_APPS and settings.GOOGLE_ANALYTICS_JS_PROPERTY_ID != '')
```

Comment:
Can we generalize this in anyway? Django-Analytical supports many analytics platforms from what I understand. Another thing to think about is that this really should be a portal setting that is set by a portal administrator, not an app developer in the app configuration. This probably needs to be refactored to be a portal wide setting.

Reply:
Thats a good comment. If i understand you right i think this is two concerns. I'll try to explain the thinking that went into these design choices.

1- should the app developer be doing anything with tracking? I agree, the app developer shouldn't have the responsibility to manage tracking settings. The app developer actually doesn't do any configuration of tracking here. This code automates including the django tags that will put the tracking scripts in each of the app pages. Because this is web based tracking, there needs to be some asynchronous code that sends data to the service provider in each page. Rather than directly add the tags to base.html, this logical check makes sure portals that dont have tracking setup can still open apps that have included the tags.

 2- why google and not be more general? We could definitely generalize but it would be more difficult to automate the setup. That would put more tasks on the portal admin and would require a longer logical check in the app to decide which tracking tags to include. Most importantly, google lets you query the analytics data through a python api so we can make an app like this one:
https://github.com/rileyhales/tethys-analytics 
This app queries the google analytics api to get usage information and creates maps, graphs, tables, etc of how specific apps get used.